src/ConstantTimeDispatcher.js

```diff
@@ -0,0 +1,62 @@
+/*
+Copyright 2017 Vector Creations Ltd
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// singleton which dispatches invocations of a given type & argument
+// rather than just a type (as per EventEmitter and Flux's dispatcher etc)
+//
+// This means you can have a single point which listens for an EventEmitter event
+// and then dispatches out to one of thousands of RoomTiles (for instance) rather than
+// having each RoomTile register for the EventEmitter event and having to 
+// iterate over all of them.
+class ConstantTimeDispatcher {
```

Comment:
What makes `ConstantTimeDispatcher` different from just using the type and arg as the key name? E.g:
```js
EventEmitter.on(`${type} ${arg}`, function() { ... });
EventEmitter.on("room !curbaf:matrix.org", function() { ... });
```

From what I can see, there is no difference at all other than sugar coating and not needing to decide on a suitable delimiter (`" "` in this case). If you want to be pedantic, `unregister` operations are more expensive in `ConstantTimeDispatcher` because it requires you to loop over all the listeners for a given `type`, whereas the `EventEmitter` form is just a hash lookup.

I'd rather we didn't re-invent the wheel on this.